homeassistant/components/openweathermap/__init__.py

```diff
@@ -34,7 +34,7 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: OpenweathermapConfigEntry
 ) -> bool:
     """Set up OpenWeatherMap as config entry."""
-    name = entry.data[CONF_NAME]
+    name = entry.title
```

Comment:
If this is going to be the name of the device, you can omit giving the name to the device, as primary devices without name will use the entry title as the device name